### ISSUE TITLE
configlet shouldn't *REQUIRE* internet access

### ISSUE THREAD

**Issue by serialhex** [exercism/configlet#683]
I had some time to do a few things with a track, but no internet access.  I had run this command the day before, but upon running it configlet crashes.  I was able to set up a hotspot really quickly but, really, it shouldn't *need* internet access for something like this. Issue a warning that 'problem-specifications' might not be up to date?  Sure!  But fail completely?  Absolutely not.

The exact error:
```
configlet sync --tests --docs --metadata --filepaths
Updating cached 'problem-specifications' data...
fatal: unable to access 'https://github.com/exercism/problem-specifications/': Could not resolve host: github.com

failed to fetch 'main' in problem-specifications directory: 'C:\Users\Welcome\AppData\Local\exercism\configlet\problem-specifications'
exec.nim(68)             execAndCheck
Error: unhandled exception:  [OSError]
```

This is but one example I can easily show, but all operations should succeed (with warnings if necessary) without internet access, unless the operation itself access the internet (like you don't need internet to `git commit` but do need it to `git pull` for instance).

**Comment by serialhex** [comment]
Just went through the docs and there is an `-o` option for offline stuff, but IMHO it should work (again, with warnings) without this flag.  Feel free to close this if the powers-that-be feel this is unnecessary.

**Comment by ee7** [comment]
Thanks for creating this issue. Sorry for the inconvenience.

> it shouldn't *need* internet access for something like this. Issue a warning that 'problem-specifications' might not be up to date? Sure! But fail completely? Absolutely not.

I think we really do want `configlet sync` to exit immediately with an error if it couldn't fetch the latest `problem-specifications` data and `--offline` was not passed. `configlet sync` is used in scripts, and will eventually be used in track CI, and in automated PRs that update exercises - it shouldn't have the potential to quietly behave differently when offline, or if there's a temporary network problem.

So I think we want the current behavior, but with a better error message. It should advise the user to pass `--offline` if they cannot update prob-specs at the current time. Does that sound OK to you?

**Comment by serialhex** [comment]
That sounds great!  It was only in looking up other things I noticed the `-o` flag, and I've been burned too many times with software that *needs* internet access for *no good reason* that I mistakenly assumed something like that wasn't there.  Thank you @ee7!

**Comment by ee7** [comment]
Great. Thanks again for reporting the problem.

I'll re-open this issue so that it'll be closed automatically when we merge #684.

> I've been burned too many times with software that _needs_ internet access for _no good reason_

Me too. Being able to use `configlet` offline is important to me.

There was already a good error message for the more obscure situation of "no prob-specs cache, but `--offline` was passed": 

```console
$ configlet sync --offline
Error: --offline was passed, but there is no cached 'problem-specifications' repo at:
  '/home/foo/.cache/exercism/configlet/problem-specifications'
Please run once without --offline to clone 'problem-specifications' to that location.

If you currently have no (or limited) network connectivity, but you do have a local
'problem-specifications' elsewhere, you can copy it to the above location and then
use it with --offline
```

But configlet didn't cover the case reported in this issue. And there are some similar errors that we'll improve in the future.
